test/jdk/sun/net/www/http/HttpClient/KeepAliveTest.java

```diff
@@ -0,0 +1,617 @@
+        // System property.
+        if (!clientScenarios[getClientScenarioNumber(scenarioNumber)].equalsIgnoreCase(NI)) {
+            System.out.println("Client Input Parsing");
+            for (String clientScenarioString : clientScenarios[getClientScenarioNumber(scenarioNumber)].split(CLIENT_SEPARATOR)) {
+                System.out.println(clientScenarioString);
+                String key = clientScenarioString.split("=")[0];
+                String value = clientScenarioString.split("=")[1];
+                System.setProperty(key, value);
+            }
+        }
+        // wait until ServerSocket moves to listening state.
+        //while (!serverReady) {
+        //    Thread.sleep(TIMEOUT);
+        //}
+        this.serverCountDownLatch.await();
+        System.out.println("client started");
+        URL url = URIBuilder.newBuilder().scheme("http").loopback().port(SERVER_PORT).toURL();
+        System.out.println("connecting from client to SERVER URL:" + url);
+        HttpURLConnection httpUrlConnection = null;
+        /*
+         * isProxySet is set to true when Expected Server Response contains Proxy-Connection header.
+         */
+        if (isProxySet) {
+            httpUrlConnection = (sun.net.www.protocol.http.HttpURLConnection) url
+                .openConnection(new Proxy(Type.HTTP, new InetSocketAddress("localhost", SERVER_PORT)));
+        } else {
+            httpUrlConnection = (sun.net.www.protocol.http.HttpURLConnection) url.openConnection();
+        }
+        InputStreamReader inputStreamReader = new InputStreamReader(httpUrlConnection.getInputStream());
+        BufferedReader bufferedReader = null;
+        try {
+            bufferedReader = new BufferedReader(inputStreamReader);
+            while (true) {
+                String eachLine = bufferedReader.readLine();
+                if (eachLine == null) {
+                    break;
+                }
+                System.out.println(eachLine);
+            }
+        } finally {
+            if (bufferedReader != null) {
+                bufferedReader.close();
+            }
+        }
+        //    System.out.println("ResponseCode:" + httpUrlConnection.getResponseCode());
+        //    System.out.println("ResponseMessage:" + httpUrlConnection.getResponseMessage());
+        //    System.out.println("Content:" + httpUrlConnection.getContent());
+        //    Thread.sleep(2000);
+        for (Entry<String, List<String>> header : httpUrlConnection.getHeaderFields().entrySet()) {
+            System.out.println(header.getKey() + "=" + header.getValue());
+        }
+        fetchInfo(scenarioNumber, httpUrlConnection);
+    }
+    private void fetchInfo(int scenarioNumber, sun.net.www.protocol.http.HttpURLConnection httpUrlConnection)
+        throws Exception {
+            Field field = Class.forName("sun.net.www.protocol.http.HttpURLConnection").getDeclaredField("http");
```

Comment:
Strange tabs here: eight spaces should be enough for a method body.

Reply:
yes removed additional spaces @dfuch thanks